roles/openshift_health_checker/openshift_checks/docker_image_availability.py

```diff
@@ -170,6 +181,9 @@ def known_docker_registries(self):
         elif deployment_type == 'openshift-enterprise' and "registry.access.redhat.com" not in regs:
             regs.append("registry.access.redhat.com")
 
+        oreg_url = self.get_var("oreg_url", default="")
+        if oreg_url != '':
+            regs.append(oreg_url.split('/')[0])
```

Comment:
Hm... I think it's possible to use `oreg_url` usefully without specifying a registry. We probably don't need to handle that though (here it would just add a registry that's not likely to be reachable). I'm slightly more concerned that this would effectively be adding a registry in the check that's not actually configured into docker. And not building the images below using `oreg_url`. Also not every image is built from `oreg_url`. May need to refine this a bit.

Reply:
I'm open to suggestion here.  My review of the code looks like oreg_url should probably be checked.  If users provide something completely nonstandard, eg. instead of 'openshift3/xxx' it's 'myorgscontainers/xxx', I would think the oreg needs to be checked to ensure the images are available because they won't be available elsewhere.

Reply:
Looking at this further, `oreg_url` is already used in generating the required images where it's relevant. There isn't a need to add to the list of registries here (which is intended to be the registries configured into docker, as docker would consult them on images not scoped by registry); if the image already includes a registry, as with `oreg_url` normally, that registry will already be used in `is_available_skopeo_image`.